b1ddi/provider.go

```diff
@@ -143,3 +159,30 @@ func dataSourceSchemaFromResource(resource func() *schema.Resource) *schema.Reso
 		Schema: resultSchema,
 	}
 }
+
+func newTransport(version string) *customTransport {
+	return &customTransport{
+		originalTransport: http.DefaultTransport,
+		terraformVersion:  version,
+	}
+}
+
+type customTransport struct {
+	originalTransport http.RoundTripper
+	terraformVersion  string
+}
+
+func (c *customTransport) RoundTrip(r *http.Request) (*http.Response, error) {
```

Comment:
@swoirgard I added the headers here. I am not sure how we can get the commit ID you wanted. I was able to get the versions easily instead.

We use `openapi/runtime` to define API clients in our `go-sdk` repo. The runtime module doesn't allow us to create custom headers but has a provision to add Auth headers. So as a workaround, I created a custom HTTP transport to add the headers.

I haven't added the `x-infoblox-pb` header in this change. We will be adding this once we move to the newer repo and figure out a way to expose this info to the provider.

Reply:
@achernevskii Has showed me the way to get commit IP, will update the code here

Reply:
* Shouldn't we use the standard `User-Agent` header value to carry these information?
* Calling `util.GetGoSDKBuild()` which returns a constant string for each HTTP request is an overkill.
* Not sure if this "workaround" is clean regarding the following from https://pkg.go.dev/net/http#RoundTripper:
```go
// RoundTrip should not modify the request, except for
// consuming and closing the Request's Body. RoundTrip may
// read fields of the request in a separate goroutine. Callers
// should not mutate or reuse the request until the Response's
// Body has been closed.
```

Reply:
As discussed, I have moved the tracking implementation out of this PR